mdata/reorder_buffer.go

```diff
@@ -79,3 +79,18 @@ func (rob *ReorderBuffer) Get() []schema.Point {
 
 	return res
 }
+
+func (rob *ReorderBuffer) Reset() {
+	for i := range rob.buf {
+		rob.buf[i].Ts = 0
+		rob.buf[i].Val = 0
+	}
+	rob.newest = 0
+}
+
+func (rob *ReorderBuffer) Flush() []schema.Point {
+	res := rob.Get()
```

Comment:
This will remove datapoints that might still need to stay in the `rob` for reordering. In my mind, `Flush` should only remove the datapoints that have aged out of the `rob`.

Reply:
Because we are now updating lastWrite when adding to the rob, flush() will always want to get all points as they will all be older then chunkMaxStale.   It is important that metrics are not delayed from being persisted for more then chunkMaxStale.  If the metrics dont get written, and the MT instance restarts the metrics could be lost, depending on the kafka retention.
To prevent loss, the kafka retention needs to be at least `(chunk-max-stale + gc-interval)`

Reply:
>Because we are now updating lastWrite when adding to the rob, flush() will always want to get all points as they will all be older then chunkMaxStale

I don't think this is true. The `gc-interval` and `chunk-max-stale` are completely independent and `Flush()` is going to be called unconditionally for every GC cycle. This means you could write data to the `rob` just 1 second before a GC cycle happens to kick off. That data is not yet ready to `Flush()` and is part of the `currentChunk`. 

If the `rob` were only `Flush()`d AFTER checking the `lastWriteTime`, then I believe you would be correct.

Reply:
Ah, I see. I missed the check of the rob.